### PR TITLE
warning: fix; matches, closures, etc..

### PR DESCRIPTION
**Pull request by Horki**
@sharkdp fix warnings, please test

### CODE REVIEW

**Comment by Horki** [comment]
@sharkdp is backward compatibility with v1.40 mandatory?

**Comment by sharkdp** [comment]
> @sharkdp is backward compatibility with v1.40 mandatory?

Not mandatory, but desirable. I'm okay with bumping the min. supported version if there is a good reason to do so.

File: src/bin/bat/app.rs [hunk 1]
```diff
@@ -82,11 +82,10 @@ impl App {
             Some("always") => PagingMode::Always,
             Some("never") => PagingMode::Never,
             Some("auto") | None => {
-                if self.matches.occurrences_of("plain") > 1 {
+                if self.matches.occurrences_of("plain") > 1
+                    || self.matches.is_present("no-paging") {
```
Comment (sharkdp) [inline src/bin/bat/app.rs:86]: I don't think that's really better. The comment does not apply to `--no-paging`.

File: src/bin/bat/app.rs [hunk 2]
```diff
@@ -169,7 +168,7 @@ impl App {
                 || match self.matches.value_of("color") {
                     Some("always") => true,
                     Some("never") => false,
-                    Some("auto") | _ => {
+                    _ => {
```
Comment (sharkdp) [inline src/bin/bat/app.rs:171]: Yes, this is the same. But I still like the previous version better.

File: src/bin/bat/app.rs [hunk 3]
```diff
@@ -226,17 +225,14 @@ impl App {
             style_components,
             syntax_mapping,
             pager: self.matches.value_of("pager"),
-            use_italic_text: match self.matches.value_of("italic-text") {
-                Some("always") => true,
-                _ => false,
-            },
+            use_italic_text: matches!(self.matches.value_of("italic-text"), Some("always")),
```
Comment (sharkdp) [inline src/bin/bat/app.rs:228]: Couldn't we simply use
```
self.matches.value_of("italic-text") == Some("always")
```
?

We're not really using the powerful feature of the `matches!` macro here.

File: src/input.rs [hunk 4]
```diff
@@ -135,17 +135,12 @@ impl<'a> Input<'a> {
     }
 
     pub fn is_stdin(&self) -> bool {
-        if let InputKind::StdIn = self.kind {
-            true
-        } else {
-            false
-        }
+        matches!(self.kind, InputKind::StdIn)
```
Comment (sharkdp) [inline src/input.rs:138]: Similar here.
```
if InputKind::StdIn = self.kind {
```
should work just fine?

File: src/pretty_printer.rs [hunk 5]
```diff
@@ -54,6 +54,12 @@ impl<'a> PrettyPrinter<'a> {
             active_style_components: ActiveStyleComponents::default(),
         }
     }
+}
+
+impl<'a> PrettyPrinter<'a> {
+    pub fn new() -> Self {
+        PrettyPrinter::default()
+    }
```
Comment (sharkdp) [inline src/pretty_printer.rs:62]: welll.. either `new` or `default`. But not both?

**Review by sharkdp — CHANGES_REQUESTED** [review]
Thank you very much for your contribution!

File: src/input.rs (hunk 4 above)
Comment (Horki) [reply src/input.rs:138]: Please set disable linters that you dont like. Like this now is confusing and annoying

Comment (sharkdp) [reply src/input.rs:138]: What do you mean? Why would we have to do that? It's not like we are enforcing zero clippy warnings anywhere.

Also, It's not that I don't like `clippy`. I think it's a great tool. But a linter does not *always* have the best suggestions. Which is why I added review comments to a few of your changes.

Comment (Horki) [reply src/input.rs:138]: @sharkdp me personally prefer the idiomatic rust way.
It does the same thing in (mostly) fewer lines of code, and there is no down|upgrade in perfomance; also warning are really annoying.

I love using this tool; and looking forward seeing new updates.

Comment (sharkdp) [reply src/input.rs:138]: > me personally prefer the idiomatic rust way

I, too, like idiomatic Rust. That's why I prefer
``` rust
     pub fn is_stdin(&self) -> bool { 
         self.kind == InputKind::StdIn
     } 
```
over
``` rust
     pub fn is_stdin(&self) -> bool { 
         matches!(self.kind, InputKind::StdIn)
     } 
```
Not only is it shorter, but also easier to read, especially for someone who doesn't know what `matches!(…)` does - which is rather new.

That's also why I prefer to have one single function that creates a `PrettyPrinter`. Either `default()`, or `new()`. Having both of them with the same functionality is not just confusing but also more code.

I added comments to a few of your changes. If you don't agree, please argue why you like your version better. Simply adding :-1: to all of my posts without giving any reasons is not really the greatest way to interact.

> also warning are really annoying

If there are compiler warnings, we should fix them, as they typically point to underlying problems. But what you are doing here is to simply copy & paste the `clippy` suggestion (*"help: try this: `matches!(self.kind, InputKind::StdIn)`"*), which is not always the best solution.

**Pull request closed without merging**